extensions/gitpod/src/remoteConnector.ts

```
@@ -379,64 +380,186 @@ export default class LocalApp extends Disposable {
+			if (action === cancel) {
+				return;
+			}
+
+			await vscode.workspace.getConfiguration('gitpod').update('host', params.gitpodHost, vscode.ConfigurationTarget.Global);
+			this.logger.info(`Updated 'gitpod.host' setting to '${params.gitpodHost}' while trying to connect to a remote workspace`);
+		}
+
+		this.logger.info('Opening remote workspace', uri.toString());
+
+		let sshDestination: string | undefined;
+		try {
+			sshDestination = await this.getWorkspaceSSHDestination(params.workspaceId, params.gitpodHost);
 		} catch (e) {
-			const seeLogs = 'See Logs';
-			vscode.window.showErrorMessage(`Failed to connect to Gitpod workspace ${params.workspaceId}: ${e}`, seeLogs).then(async result => {
-				if (result !== seeLogs) {
-					return;
-				}
-				this.logger.show();
-				if (resolvedConfig) {
-					const document = await vscode.workspace.openTextDocument(vscode.Uri.file(resolvedConfig.logPath));
-					vscode.window.showTextDocument(document);
-				}
-			});
-			this.logger.error(`failed to open uri: ${e}`);
-			throw e;
+			if (e instanceof Error && e.message === 'no_running_instance') {
+				vscode.window.showErrorMessage(`Failed to connect to remote workspace: No running instance for '${params.workspaceId}'`);
+				return;
+			}
 		}
+
+		const usingSSHGateway = !!sshDestination;
+		let localAppSSHConfigPath: string | undefined;
+		if (!usingSSHGateway) {
+			vscode.window.showInformationMessage(`Falling back connecting to remote workspace using local app. You should update your Gitpod installation [enabling direct SSH access](https://github.com/gitpod-io/gitpod/blob/main/install/installer/docs/workspace-ssh-access.md)`);
+			try {
+				const localAppDestData = await this.getWorkspaceLocalAppSSHDestination(params);
+				sshDestination = localAppDestData.localAppSSHDest;
+				localAppSSHConfigPath = localAppDestData.localAppSSHConfigPath;
+			} catch (e) {
+				return;
```

swallowing errors? or it never can hapepn and try-catch can be removed?

---

this is fine it will only throw is user cancelled it, I'll add a comment